crates/bevy_time/src/timer.rs

```diff
@@ -14,19 +14,54 @@ use bevy_utils::Duration;
 #[reflect(Default)]
 pub struct Timer {
     stopwatch: Stopwatch,
-    duration: Duration,
+    non_zero_duration: NonZeroDuration,
     mode: TimerMode,
     finished: bool,
     times_finished_this_tick: u32,
 }
 
+/// A new-typed [`Duration`] measure of elaspsed time that is never [`Duration::ZERO`].
+///
+/// Interally, this snsures that `duration` is never zero when being used in [`Timer`]. This simplifies edge cases
+/// and removes the need for repeated runtime checks for zero duration in [Timer::tick](Timer::tick)
+#[derive(Clone, Debug, PartialEq, Eq, Reflect, FromReflect)]
+#[cfg_attr(feature = "serialize", derive(serde::Deserialize, serde::Serialize))]
+#[reflect(Default)]
+pub struct NonZeroDuration {
+    duration: Duration,
+}
+
+impl Default for NonZeroDuration {
+    fn default() -> Self {
+        Self {
+            duration: Duration::new(1, 0),
+        }
+    }
+}
+
+impl NonZeroDuration {
```

Comment:
I think this impl should have a `get` or `value` or some other method for accessing the inner duration (similar to the NonZero*** types)